### ISSUE TITLE
Benchmark Nightwatch sitemap a11y tests

### ISSUE THREAD

**Issue by billfienberg** [department-of-veterans-affairs/va.gov-team#22443]
>Does converting the Nightwatch sitemap a11y tests to Cypress get us any speed? -  https://github.com/department-of-veterans-affairs/va.gov-team/issues/22163#issuecomment-807314901

**Comment by billfienberg** [comment]
I don't think the current "parallelization" of the Nightwatch a11y tests is actually doing anything. 

```
[2021-04-05T22:22:52.417Z] http://vets-website:3001/eastern-oklahoma-health-care/operating-status/
[2021-04-05T22:22:52.671Z]  [0;32m✔[0m Element <body> was visible after 29 milliseconds.
[2021-04-05T22:22:53.229Z] 
[2021-04-05T22:22:53.229Z] [0;32mOK.[0m [0;32m885[0m assertions passed. (16m 29s / 989583ms)
[2021-04-05T22:22:53.229Z] 
[2021-04-05T22:22:53.229Z] [0;36m[Platform / Site Wide / Tests / Sitemap / Sitemap 2 Spec] Test Suite[0m
[2021-04-05T22:22:53.229Z] [0;35m========================================================================[0m
[2021-04-05T22:22:53.229Z] 
[2021-04-05T22:22:53.229Z] Running:  [0;32msitemap 2/4[0m
[2021-04-05T22:22:53.788Z] http://vets-website:3001/asheville-health-care/operating-status/
[2021-04-05T22:22:54.346Z]  [0;32m✔[0m Element <body> was visible after 21 milliseconds.
```

http://jenkins.vfs.va.gov/blue/rest/organizations/jenkins/pipelines/testing/pipelines/vets-website/branches/master/runs/10683/nodes/152/steps/155/log/?start=0

**Comment by pjhill** [comment]
This is deprioritized below research into a quick parallelization of the Cypress E2E tests as that is now the longest running task in the build pipeline.

**Comment by ddzz** [comment]
WIP branch with tests running in parallel on GitHub Actions: https://github.com/department-of-veterans-affairs/content-build/compare/migrate-a11y-tests-gha?expand=1

**Comment by 1Copenut** [comment]
Worked on the local test runner. Trying to recreate a situation where one or more tests fail on async events like map loading.

**Comment by 1Copenut** [comment]
Tried to get the map running locally. Hitting API roadblocks, will pick it up Monday morning.

**Comment by 1Copenut** [comment]
Spent most of yesterday working on the nuances of the a11y tests. We have what I believe is a good conditional check for pages with an async map being loaded. There are a number of pages where the data call to load the map fails, so I'm logging those with the relevant team today.

**Comment by 1Copenut** [comment]
Worked some more on the a11y sitemap tests. We've got a high number of errors in my conditional logic to wait for maps to load. Darius and I are looking into it--most of them appear to be false positives.

**Comment by 1Copenut** [comment]
Running the entire suite locally (taking all day) to test a new theory. Trying out ignoring the async map loads altogether and just focusing on the static content page. It's not without risk, but does make the tests run more consistently.

**Comment by 1Copenut** [comment]
Got the tests running cleanly on our feature branch. They're still failing in the main GitHub Action, but passing in the Accessibility Action, so we're getting closer.

**Comment by 1Copenut** [comment]
Latest test run is very close to running stable. I'm seeing two a11y tests fail on the Install dependencies step, and two that are timing out on the axe check. Guessing the timeouts are because of a modal window opening on the pages. Might see if we can ignore those pages or work around in the `axeCheck` configuration settings.

**Comment by 1Copenut** [comment]
Hoping Darius can help me get this across the finish line on Monday.

**Comment by pjhill** [comment]
The scan is running and reporting results. We do want to make enhancements to the reporting which will be worked under a new ticket [here](https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/29913).
